addons/website_blog/controllers/main.py

```diff
@@ -184,11 +184,12 @@ def blog(self, blog=None, tag=None, page=1, search=None, **opt):
 
         blogs = Blog.search(request.website.website_domain(), order="create_date asc, id asc")
 
-        if not blog and len(blogs) == 1:
-            return werkzeug.utils.redirect('/blog/%s' % slug(blogs[0]), code=302)
-
         date_begin, date_end, state = opt.get('date_begin'), opt.get('date_end'), opt.get('state')
 
+        if not blog and len(blogs) == 1:
+            url = QueryURL('/blog/%s' % slug(blogs[0]), date_begin=date_begin, date_end=date_end, search=search)()
```

Comment:
As discussed: use **opt (needed to keep order too for example)

Reply:
Code & commit msg adapted :+1: .
Note that the order param is not even working on blog, it's a capability added by the generic search snippet which do not work on blogs it seems.. We should probably do a full testing of this app at some point.
But it's better to fix it properly indeed, a reproductible case playing with the URL is with the `state` param (you can pass `published`).